posthog/temporal/data_imports/workflow_activities/import_data_sync.py

```diff
@@ -255,6 +282,26 @@ def import_data_activity_sync(inputs: ImportDataActivityInputs):
             raise ValueError(f"Source type {model.pipeline.source_type} not supported")
 
 
+def _handle_non_retryable_error(
+    job_inputs: PipelineInputs, error_msg: str, logger: FilteringBoundLogger, error: Exception
+) -> NoReturn:
+    with _get_redis() as redis_client:
+        if redis_client is None:
+            logger.debug(f"Failed to get Redis client for non-retryable error tracking. error={error_msg}")
+            raise NonRetryableException() from error
+
+        retry_key = _non_retryable_errors_key(job_inputs)
+        attempts = redis_client.incr(retry_key)
+
+        if attempts <= 3:
+            redis_client.expire(retry_key, 600)  # Expire after 10 minutes
+            logger.debug(f"Non-retryable error attempt {attempts}/3, retrying. error={error_msg}")
+            raise
+
+    logger.debug(f"Non-retryable error after {attempts-1} attempts, giving up. error={error_msg}")
```

Comment:
This will log `Non-retryable error after 3 attempts` when really it'll be the 4th attempt?